### PR TITLE
Add initial implementation of Table's BIR Model

### PR DESCRIPTION
**Pull request by gimantha**
## Purpose
This PR contains the implementation which generates the BIR of Ballerina tables. The PR will add the model classes and the visitor implementation to generate the BIR of Table

## Approach
Introduce the NewTable instruction in BIR generation phase to create a new table. Along with that, the primary keys indexes and the initial data set is passed.
The instruction looks like below.

`target_Address = new_table [columns, indices, keys, data]`

- columns is the location of columns array.
- indices is the location of indices array.
- keys is the location of primary keys array.
- data is the location of initial data array of records.

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/ballerina-unit-test/src/test/resources/test-src/bir/bir_model.bal [hunk 1]
```diff
@@ -112,16 +112,33 @@ function dump(int arg1, int arg2) returns int {
...
+    table<Employee> employeeTable = table {
+        {id, name},
+        [
+            {1, "Employee1"},
+            {2, "Employee2"}
+        ]
+    };
+
+    return employeeTable;
+}
```
Comment (Kishanthan) [inline tests/ballerina-unit-test/src/test/resources/test-src/bir/bir_model.bal]: new line missing

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRInstructionWriter.java [hunk 2]
```diff
@@ -291,6 +291,16 @@ public void visit(BIRNonTerminator.TypeTest birTypeTest) {
         birTypeTest.rhsOp.accept(this);
     }
 
+    public void visit(BIRNonTerminator.NewTable newTable) {
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRInstructionWriter.java]: Don't we need to write position here?

Comment (gimantha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRInstructionWriter.java]: Its added

File: tests/ballerina-unit-test/src/test/resources/test-src/bir/bir_model.bal (hunk 1 above)
Comment (gimantha) [reply tests/ballerina-unit-test/src/test/resources/test-src/bir/bir_model.bal]: Added

**Pull request merged**